subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -155,6 +165,35 @@ int z_impl_z_zsock_getaddrinfo_internal(const char *host, const char *service,
 		return getaddrinfo_null_host(port, hints, res);
 	}
 
+#define SIN_ADDR(ptr) (((struct sockaddr_in *)(ptr))->sin_addr)
+
+	/* Check for IPv4 numeric address. Start with a quick heuristic check,
+	 * of first char of the address, then do long validating inet_pton()
+	 * call if needed.
+	 */
+	if (RESOLVE_IPV4(family)
+	    && isdigit((int)*host)
```

Comment:
the && goes at the end of the last line. Apply everywhere.

Reply:
fixed